pyglet/util.py

```diff
@@ -16,25 +18,40 @@ def asbytes(s):
         return bytes(s)
 
 
-def asbytes_filename(s):
-    if isinstance(s, bytes):
+def asbytes_filename(s: Union[str, Buffer]) -> Optional[bytes]:
+    if isinstance(s, (bytes, bytearray, memoryview)):
         return s
     elif isinstance(s, str):
         return s.encode(encoding=sys.getfilesystemencoding())
 
+    return None
 
-def asstr(s):
+def asstr(s: Optional[Union[str, Buffer]]) -> str:
     if s is None:
         return ''
     if isinstance(s, str):
         return s
-    return s.decode("utf-8")
+    return s.decode("utf-8")  # type: ignore
+
+
+# Keep these outside of the function since we don't need to re-define
+# the function each time we make a call since no state is persisted.
+def _debug_print_real(arg: str) -> bool:
+    print(arg)
+    return True
+
+
+def _debug_print_dummy(arg: str) -> bool:
+    return True
+
 
+def debug_print(pyglet_option_name: str = 'debug') -> DebugPrintCallable:
+    """Get a debug print callable based on a pyglet option name.
 
-def debug_print(enabled_or_option='debug'):
-    """Get a debug printer that is enabled based on a boolean input or a pyglet option.
-    The debug print function returned should be used in an assert. This way it can be
-    optimized out when running python with the -O flag.
+    The debug print function returned should be used in an assert so
+    it can automatically be removed when running Python with the -O
+    flag. Both of the returned functions return True to ensure the
+    types for this work.
 
     Usage example::
 
```

Comment:
I just noticed this, but there is no such thing as `pyglet.debug`. :sweat:

Reply:
Good catch, ty. I've fixed it for now, but we might want to rethink some this design in the future. More thoughts on that later.